src/api.rs

```diff
@@ -602,37 +619,15 @@ impl Context {
 
         let sef_data = encode_frame(fi, &mut fs);
         self.packet_data.extend(sef_data);
-        let data = self.packet_data.clone();
-        self.packet_data.clear();
-        if write_temporal_delimiter(&mut self.packet_data).is_err() {
-          return Err(EncoderStatus::Failure);
-        };
 
         let rec = if fi.show_frame { Some(fs.rec) } else { None };
-        let mut psnr = None;
-        if self.config.enc.show_psnr {
-          if let Some(ref rec) = rec {
-            psnr = Some(calculate_frame_psnr(
-              &*fs.input,
-              rec,
-              fi.sequence.bit_depth
-            ));
-          }
-        }
-
-        self.frames_processed += 1;
-        Ok(Packet {
-          data,
-          rec,
-          number: fi.number,
-          frame_type: fi.frame_type,
-          psnr
-        })
+        let fi = fi.clone();
+        self.finalize_frame(&*fs.input, rec, &fi)
```

Comment:
this would be nicer in a separate patch.

Reply:
Opened https://github.com/xiph/rav1e/pull/982

Reply:
Nice, now it needs just a rebase :)